src/plugins/data/common/index_patterns/index_patterns/types.ts

```diff
@@ -33,3 +33,16 @@ export interface TypeMeta {
   aggs?: Record<string, AggregationRestrictions>;
   [key: string]: any;
 }
+
+export interface IndexPatternSpec {
+  _id: string;
+  _type: string;
+  _source: {
+    title: string;
```

Comment:
why are this under _source ?

Reply:
feels like we might just want whats under _source ? could we restore index pattern withoiut the other stuff ?

Reply:
@ppisljar I agree that we could do that and it would likely be sufficient for your needs. However, that would be a different `IndexPattern` - one that couldn't be saved. I'd prefer to have just one kind of index pattern. We could consider having different types after some of the planned refactoring.

Reply:
what do you mean it couldn't be saved ? if
```
export interface IndexPatternSpec {
  id: string;
  type: string;
  title: string;
  timeFieldName: string;
  fields: Field[];
  version: string;
}
}
```

and then `specToIndexPattern` takes this in and gives you back indexPattern instance that can be saved.

lets try to make this IndexPatternSpec a bit easier to handle and more futureproof if possible. the information on how saved object looks like does not need to be exposed to the consumer (IndexPatternSpec can have a different shape as long as it contains sufficient information to reconstruct index pattern out of it)

Reply:
version can also be removed as that's a saved object thing

Reply:
I guess I didn't understand what you were asking for, happy to implement `IndexPatternSpec` as explained above. 

That said, I think we need to keep `version` as its used when updating an index pattern.

Reply:
we should always assume the latest version when creating index pattern from spec.

Reply:
version is used to ensure there aren't write collisions.